genie/validate.py

```diff
@@ -415,3 +416,96 @@ def standardize_string_for_validation(
         return standardized_str
     else:
         return input_string
+
+
+def get_invalid_allele_rows(
+    input_data: pd.DataFrame,
+    input_col: str,
+    allowed_comb_alleles: list,
+    allowed_ind_alleles: list,
+    ignore_case: bool = False,
+    allow_na: bool = False,
+) -> pd.Index:
+    """
+    Find invalid indices in a DataFrame column based on allowed allele values.
+
+    Args:
+        input_data (pd.DataFrame): The DataFrame to search.
+        input_col (str): The name of the column to check.
+        allowed_comb_alleles (list): The list of allowed allele values
+            (can appear in combinations or individually)
+        allowed_ind_alleles (list): The list of allowed allele values
+            (can only appear individually)
+        ignore_case (bool, optional): whether to perform case-insensitive matching
+        allow_na (bool, optional): whether to allow NAs to be an allowed allele
+            value or not.
+    Returns:
+        pd.Index: A pandas index object indicating the row indices that
+        don't match the allowed alleles
+    """
+    search_str = ""
+    if allowed_comb_alleles:
+        search_str += f'^[{re.escape("".join(allowed_comb_alleles))}]+$'
+
+    if allowed_ind_alleles:
+        search_str += f'|^[{re.escape("".join(allowed_ind_alleles))}]+$'
+
+    if ignore_case:
+        flags = re.IGNORECASE
+    else:
+        flags = 0  # no flags
+
+    # special handling for all NA column
+    is_all_na = pd.isna(input_data[input_col]).all()
+    if is_all_na and allow_na:
```

Comment:
Not the biggest fan of what has to be done here to allow the checking of non-str columns and columns with all NAs. I almost feel like it would be better to have the `_check_allele_col` be an enforced rule where it skips certain validation rules if the column has any NAs OR if we choose to allow NAs, have a standardized handling for those (if that's even possible...) so that we don't have to do it in the individual validation functions.

Reply:
Agreed, can you create a ticket to track that?  for now, let's do the 80/20 rule.  This does exactly what we need it to do.